odoo_backup_sh/models/odoo_backup_sh.py

```diff
@@ -235,14 +267,21 @@ def create_info_file(self, info_file_object, storage_service):
     def delete_remote_objects(self, cloud_params, remote_objects):
         odoo_backup_sh_objects = []
         for file in remote_objects:
+            # ignore deletion of objects from the storage server if the file is a simulation object
+            file_name = file[0]
+            file_name_parts = file_name.split('.')
+            backup_dt_part_index = -3 if file_name_parts[-1] == 'enc' else -2
+            db_name = '.'.join(file_name_parts[:backup_dt_part_index])
+            config = self.search([('database', '=', db_name), ('storage_service', '=', 'odoo_backup_sh')])
+            if config.backup_simulation:
+                continue
```

Comment:
Looks complicated. Why just don't you change calling ``delete_remote_objects`` instead (e.g. don't add simulated backups to ``remote_objects_to_delete``)?